ingeniamotion/fsoe_master/handler.py

```diff
@@ -418,9 +442,13 @@ def remove_pdo_maps_from_slave(self) -> None:
         self.__servo.remove_tpdo_map(self.safety_slave_pdu_map)
 
     def get_request(self) -> None:
-        """Set the FSoE master handler request to the Safety Master PDU PDOMap."""
+        """Set the FSoE master handler request to the Safety Master PDU PDOMap.
+
+        Raises:
+            RuntimeError: If the FSoE Master is not running.
+        """
         if not self.__running:
-            self.__start_on_first_request()
+            raise RuntimeError("FSoE Master is not running")
```

Comment:
does the pdo thread have some kind of exception contention? This looks a bit hard.
not sure if a race condition could trigger this

Reply:
In fact the fw does have this kind of condition. The fsoe slave starts a bit later than the ethercat pdo slave so there's a few cycles where the map is still empty

Reply:
Just a concern, maybe we just have to handle this problem on anther layer if it ever comes.
We have talked about decoupling the fsoe to another thread, which could be a source of this. We'll just have to be extra careful